### DISCUSSION TITLE
Could Alembic use SQLite3's column renaming support?

### DISCUSSION THREAD

**Discussion by robertlagrant** [sqlalchemy/alembic#1576] [Ideas]
[SQLite 3.25.0](https://www.sqlite.org/changes.html#version_3_25_0) has support for renaming table columns.

It would be good to replace the clever but painful method Alembic uses to rename SQLite columns with this, as it means that things like constraints set on the table no longer have to be removed and re-added in migrations.

It is possible to check the version of SQLite present - 

In Python:
```python
>>> import sqlite3
>>> sqlite3.sqlite_version
'3.47.0'
```

Or in SQL:
```sql
SELECT sqlite_version();
```

This way perhaps both methods could be supported in parallel.

**Comment by zzzeek** [comment]
id rather wait for SQLite to just support ALTER   (which they most certainly can, despite sqlite's unusual storage model)

**Reply by robertlagrant** [reply]
What are you after? This uses ` ALTER TABLE table RENAME COLUMN oldname TO newname`

**Reply by zzzeek** [reply]
our add_column() I think already uses SQLite's normal ALTER facility and you can use alter_column with SQLite right now to do a rename column, I guess it needs the keyword COLUMN?

```
    def test_rename_column(self):
        context = op_fixture("sqlite")
        op.alter_column("t1", column_name="old", new_column_name="new")
        context.assert_("ALTER TABLE t1 RENAME old TO new")

```

**Reply by zzzeek** [reply]
but otherwise no, Alembic wouldnt do any version detection stuff, just use the construct or not basically

**Reply by zzzeek** [reply]
https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/5573

**Reply by zzzeek** [reply]
it's merged it will be in the next release

**Reply by robertlagrant** [reply]
Thanks!

**Comment by vkama** [comment]
I am using 1.14.1 but on a column rename it still drops, then creates a column instead of using alter_table.
Anything extra I need to do or should I edit the migration manually?

thanks

**Reply by CaselIT** [reply]
You need to edit the migration manually, alembic autogenerate doesn't generate column renames automatically at this time
